Fix Exec and ToolTask for Linux and minor text fixes

ValMenn opened the pull request:
1. Make sure we quote string that follows "sh -c" in Exec and ToolTask
2. If there is nothing following "sh -c", Unix shell
   becomes interactive. Prevent it by adding an 'echo'
   to the command.
3. Fix encoding environment variables in Exec task
4. Minor text fixes related to casing and slashes

AndyGerlicher commented:
LGTM

rainersigwald commented on src/Utilities/UnitTests/ToolLocationHelper_Tests.cs:
```diff
@@ -57,7 +57,7 @@ public void GetApiContractReferencesHandlesNonExistingLocation()
         {
             string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
-            string referenceDirectory = Path.Combine(tempDirectory, @"References\Foo\Bar");
+            string referenceDirectory = Path.Combine(tempDirectory, NativeMethodsShared.IsWindows ? @"References\Foo\Bar" : "References/Foo/Bar");
```
`Path.Combine(tempDirectory, "References", "Foo", "Bar")` should work everywhere, right?

rainersigwald commented on src/Utilities/UnitTests/ToolLocationHelper_Tests.cs:
```diff
@@ -380,6 +380,12 @@ public void FindFrameworksPathRunningThisTest()
         [Fact]
         public void FindFrameworksPathRunningUnderWhidbey()
         {
+            if (!NativeMethodsShared.IsWindows)
```
I think Roslyn does this with a test trait, which might be a better idea.  We should consider that in the future (not now)

rainersigwald commented on src/Utilities/UnitTests/ToolTask_Tests.cs:
```diff
@@ -31,7 +31,7 @@ public MyTool()
                     FileUtilities.GetFolderPath(FileUtilities.SpecialFolder.System),
 #endif
-                    NativeMethodsShared.IsUnixLike ? "/bin/sh" : "cmd.exe");
+                    NativeMethodsShared.IsUnixLike ? "sh" : "cmd.exe");
```
Can you add a note in the commit message about why you're switching away from `/bin/`?

rainersigwald commented on src/Utilities/UnitTests/ToolTask_Tests.cs:
```diff
@@ -222,7 +227,7 @@ public void HandleExecutionErrorsWhenToolLogsError()
                 t.MockCommandLineCommands = NativeMethodsShared.IsWindows
                                                 ? "/C echo Main.cs(17,20): error CS0168: The variable 'foo' is declared but never used"
-                    : @"-c ""echo \""Main.cs(17,20): error CS0168: The variable 'foo' is declared but never used\""""";
+                                                : @"-c """"""echo Main.cs\(17,20\): error CS0168: The variable 'foo' is declared but never used""""""";
```
So. Many. Quotes :fearful:

rainersigwald commented:
LGTM

ValMenn commented:
Addressed comments and rebased. The **/bin/sh** to **sh** question was addressed in commit message (basically, since SpecialFolder.System now returns **/bin**, we ended up with **/bin/bin/sh**).
